### ISSUE TITLE
ImageAnalysisErrorReason.CONNECTION_FAILURE

### ISSUE THREAD

**Issue by lordlinus** [Azure-Samples/azure-ai-vision-sdk#40]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [X ] bug report -> please search issues before submitting
- [ ] feature request
- [X ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
>
run the sample code from the documentation page at https://github.com/Azure-Samples/azure-ai-vision-sdk/blob/main/docs/learn.microsoft.com/python/image-analysis/3/main.py
endpoint used format - `https://{name_of_cogs_service}.cognitiveservices.azure.com/`
valid api key is used - the below curl request from the same pc works fine

curl -X POST "https://{name_of_cogs_service}.cognitiveservices.azure.com/computervision/imageanalysis:analyze?api-version=2023-02-01-preview&features=objects&language=en&gender-neutral-caption=False" \
    -H "Content-Type: application/json" -H "Ocp-Apim-Subscription-Key: VALID_KEY_USED" \
    --data "{\"url\":\"https://images.openai.com/blob/9ad5a2ab-041f-475f-ad6a-b51899c50182/ingredients.png\"}"

### Any log messages given by the failure
>
 Analysis failed.
   Error reason: ImageAnalysisErrorReason.CONNECTION_FAILURE
   Error code: -1
   Error message: Unknown error in sending http request

### Expected/desired behavior
>
output should be json with the below result 
{"objectsResult":{"values":[{"boundingBox":{"x":167,"y":419,"w":301,"h":279},"tags":[{"name":"Egg","confidence":0.543}]},{"boundingBox":{"x":247,"y":669,"w":291,"h":285},"tags":[{"name":"Egg","confidence":0.617}]},{"boundingBox":{"x":445,"y":405,"w":559,"h":358},"tags":[{"name":"Food","confidence":0.711}]}]},"modelVersion":"2023-02-01-preview","metadata":{"width":1006,"height":1006}}
### OS and Version?
> Windows 7, 8 or 10. Linux (which distribution). macOS (Yosemite? El Capitan? Sierra?)

### Versions
>
azure-ai-vision-0.13.0b1
python 3.10.10

### Mention any other details that might be useful

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by mattmazzola** [comment]
I can confirm I also receive this error. I hadn't been able to receive a single successful request using the 4.0 API.

The "unknown error" is not very helpful. I am also suspicious because I am using HTTP**S** but the error message says HTTP.

azure-ai-vision: 0.13.0b1
python: 3.10.6

- Relevant Docs References
  - [Error Codes](https://learn.microsoft.com/en-us/azure/cognitive-services/computer-vision/how-to/call-analyze-image-40?tabs=rest#error-codes)
  - [ImageAnalysisErrorReason Fields](https://learn.microsoft.com/en-us/python/api/azure-ai-vision/azure.ai.vision.enums.imageanalysiserrorreason?view=azure-python-preview#fields)

**Comment by dargilco** [comment]
@lordlinus please run sample number 0 (`image_analysis_sample_analyze`) here: https://github.com/Azure-Samples/azure-ai-vision-sdk/tree/main/samples/python/image-analysis . Your link points to an example of using custom model. Seems like you did not intend to use a custom model.

**Comment by dargilco** [comment]
@mattmazzola  please share more information on your case. Share source code. What OS/Platform is this? Does an equivalent CURL command work? What happens when you run sample number 0 here with your endpoint and computer vision key? https://github.com/Azure-Samples/azure-ai-vision-sdk/tree/main/samples/python/image-analysis

**Comment by lordlinus** [comment]
> 

@dargilco i get the same error when i ran the sample 

![image](https://github.com/Azure-Samples/azure-ai-vision-sdk/assets/2416507/f453aecb-bc0b-49c9-83e4-54803d7543ea)

**Comment by dargilco** [comment]
@lordlinus this error message indicates that Open SSL 1.x was not installed. Per the [README](https://github.com/Azure-Samples/azure-ai-vision-sdk/tree/main/samples/python/image-analysis), On Ubuntu 22.04 LTS it is also required to download and install the latest libssl1.1 package (e.g. from http://security.ubuntu.com/ubuntu/pool/main/o/openssl/). Ubuntu 22.04 comes with OpenSSL 3.x, and no longer with OpenSSL 1.x. The Vision SDK has not yet been updated to support OpenSSL 3.x. We plan to fix this in one of the next releases.

**Comment by mattmazzola** [comment]
> please share more information on your case

I created a video to give some more context. I can't easily share the code without creating separate repo and moving my script and dependencies over to make it public.

https://github.com/Azure-Samples/azure-ai-vision-sdk/assets/2856501/34d7f777-b669-4598-8c68-e8ef8b34008f

> this error message indicates that Open SSL 1.x was not installed

You may know that from internal knowledge of how the code works, but this error message does NOT imply that is the issue.
It only says "Unknown error in sending http request"

It seems like a generic error for any issues that happen. 
It would be more helpful if you listed probable causes for that error in the error message such as:

"If you are using Ubuntu 22.04 required to download and install the latest libssl1.1 package (e.g. from http://security.ubuntu.com/ubuntu/pool/main/o/openssl/)"

I suspect I am having the same problem since I am running from within this container environment:

`mcr.microsoft.com/devcontainers/base:ubuntu-22.04`

**Comment by mattmazzola** [comment]
> let me know if that solved your problem

I was struggling with the manual installation steps and I ended up downgrading the container to use `mcr.microsoft.com/devcontainers/base:ubuntu-20.04` and now I can use the Python SDK with v4.0 API and am unblocked! 😄Thank for the response about Open SSL.

**Comment by dargilco** [comment]
I added [this document](https://github.com/Azure-Samples/azure-ai-vision-sdk/blob/main/docs/ubuntu2204-notes.md) to the samples repo, and updated the relevant Linux README files to point to it. I also made an SDK code change such that the error message will mention that if you are running on Ubuntu 22.04 you need to install OpenSSL 1.x from source code, including a link to this new document. This change will be included in the next version of the SDK (we release every 1-2 months). We will also write a dedicated Vision SDK setup page under the [public Image Analysis document](https://learn.microsoft.com/azure/cognitive-services/computer-vision/overview-image-analysis?tabs=4-0) addressing this.

@mattmazzola if the instructions to setup OpenSSL 1.x from source code did not work for you, please open a new GitHub issue with a description of what you’ve done and the issue you encountered.

@mattmazzola @lordlinus if you have any other issues or feedback in general on the SDK, feel free to open a new issue. I will close this one.

**Comment by fengjihua** [comment]
Error reason: ImageAnalysisErrorReason.CONNECTION_FAILURE
Error code: -1
Error message: Unknown error in sending http request

OS and Version
CentOS 7.9.2009 (works fine on local Windows 10)

Versions
azure-ai-vision 0.13.0b1
python 3.8.12 (Gunicorn + Flask)
OpenSSL 1.0.2k
urlib 1.26


Azure-Cognitive-Speech and Azure-Cognitive-Language service are working fine on same server

**Comment by dargilco** [comment]
@fengjihua Unfortunately we do not support CentOS 7. I believe it should work on CentOS 8.
